### PR TITLE
Bump Log4j from 2.17.0 to 2.17.2

### PR DESCRIPTION
**Pull request by garydgregory**
Bump Log4j from 2.17.0 to 2.17.2.

### CODE REVIEW

File: instrumentation/log4j/log4j-appender-2.17/javaagent/build.gradle.kts [hunk 1]
```diff
@@ -14,7 +14,7 @@ muzzle {
 val testLatestDeps = findProperty("testLatestDeps") as Boolean
 
 dependencies {
-  library("org.apache.logging.log4j:log4j-core:2.17.0")
+  library("org.apache.logging.log4j:log4j-core:2.17.2")
```
Comment (laurit) [inline instrumentation/log4j/log4j-appender-2.17/javaagent/build.gradle.kts:17]: we usually compile and test agains the earliest supported version (`library` translates to `compileOnly` + `testImplementation`)

File: instrumentation/log4j/log4j-context-data/log4j-context-data-2.17/javaagent/build.gradle.kts [hunk 2]
```diff
@@ -6,13 +6,13 @@ muzzle {
   pass {
     group.set("org.apache.logging.log4j")
     module.set("log4j-core")
-    versions.set("[2.17.0,)")
+    versions.set("[2.17.2,)")
```
[the remark this answers is not in the document]
Comment (garydgregory) [reply instrumentation/log4j/log4j-context-data/log4j-context-data-2.17/javaagent/build.gradle.kts:9]: Hi @laurit 
Are you saying that this failure is due to a bug in Log4j, opentelemetry-java-instrumentation, or in the tests themselves, but we do not know which one?

File: instrumentation/log4j/log4j-appender-2.17/javaagent/build.gradle.kts (hunk 1 above)
Comment (garydgregory) [reply instrumentation/log4j/log4j-appender-2.17/javaagent/build.gradle.kts:17]: @laurit 
Thank you for explaining this. This makes it sounds like this PR is misguided; or am I missing something?

File: instrumentation/log4j/log4j-context-data/log4j-context-data-2.17/javaagent/build.gradle.kts (hunk 2 above)
[the remark this answers is not in the document]
Comment (steverao) [reply instrumentation/log4j/log4j-context-data/log4j-context-data-2.17/javaagent/build.gradle.kts:9]: > Hi @laurit Are you saying that this failure is due to a bug in Log4j, opentelemetry-java-instrumentation, or in the tests themselves, but we do not know which one?

What he means is that this change is wrong and will cause CI to run failed. You can see relevant log:  https://github.com/open-telemetry/opentelemetry-java-instrumentation/actions/runs/8737558469/job/23974806236?pr=11169#step:5:317
For moro explanation about muzzle check you can refer to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/muzzle.md

**Review by steverao — COMMENTED** [review]
Can you explain more about why do you need to adjust the version of `log4j-core`? We usually use earliest supported version here, It means the instrumentation supports from `2.17.0`. For more explanation about relevant it you can refer to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/writing-instrumentation.md#folder-structure

File: instrumentation/log4j/log4j-appender-2.17/javaagent/build.gradle.kts (hunk 1 above)
Comment (laurit) [reply instrumentation/log4j/log4j-appender-2.17/javaagent/build.gradle.kts:17]: > This makes it sounds like this PR is misguided

correct

**Comment by garydgregory** [comment]
I thought it best to test against the latest version but since this seems to be hard-wired in the name to 2.17 instead of 2.x, I picked the latest 2.17.x. This is a misunderstanding on my part regarding how this project manages dependencies. No big deal. I see what you are doing now. Thank you for taking the time to review and comment :-)

**Comment by garydgregory** [comment]
Ah, thank you for pointing that out @laurit !

**Pull request closed without merging**